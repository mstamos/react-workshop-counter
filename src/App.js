import React, {useState} from 'react';
import PropTypes from 'prop-types';
import './App.css';

export function Buttons({increase, decrease}) {
  return (
    <div>
      <button data-testid="plus" onClick={increase}>+</button>
      <button data-testid="minus" onClick={decrease}>-</button>
    </div> 
  )
}

Buttons.propTypes = {
  increase: PropTypes.func,
  decrease: PropTypes.func
}

function App() {
  const [counter, setCounter] = useState(0);

  const increase = () => {
    setCounter(counter + 1);
  }

  const decrease = () => {
    setCounter(counter - 1);
  }

  if (counter < 0) {
    return (
      <div className="App">
        <div data-testid="counter-result">:-)</div>
        <Buttons increase={increase} decrease={decrease}/>
      </div>
    )
  }

    return (
      <div className="App">
        <div data-testid="counter-result">Counter: {counter} </div>
        <Buttons increase={increase} decrease={decrease}/>
      </div> 
    )

}

// class App extends React.Component {
//   constructor() {
//     super();
//     this.state= {
//       counter: 0
//     }
//   }

//   increase = () => {
//     this.setState({
//       counter: this.state.counter + 1
//     });
//   }

//   decrease = () => {
//     this.setState({
//       counter: this.state.counter - 1
//     });
//   }

//   render() {
//     return (
//     <div className="App">
//       <div>Counter: {this.state.counter}</div>
//       <div>
//         <button onClick={this.increase}>+</button>
//         <button onClick={this.decrease}>-</button>
//       </div>
//     </div> 
//     )
//   }

// }


// function App() {
//   return (
//     <div className="App">
//       <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />
//         <p>
//           Edit <code>src/App.js</code> and save to reload.
//         </p>
//         <a
//           className="App-link"
//           href="https://reactjs.org"
//           target="_blank"
//           rel="noopener noreferrer"
//         >
//           Learn React
//         </a>
//       </header>
//     </div>
//   );
// }

export default App;
