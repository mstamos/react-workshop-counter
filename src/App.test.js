import { render, screen } from '@testing-library/react';
import App, {Buttons} from './App';

describe('App', () => {

  describe('Buttons', () => {

    it('renders', () => {
      expect(render(<Buttons/>).container).toMatchSnapshot()
    });

    it('calls increase function', () => {
      const increase = jest.fn();

      render(<Buttons increase={increase}/>)

      screen.getByTestId('plus').click()

      expect(increase).toHaveBeenCalled();
    })

  });

  it('increases the counter correctly', () => {
    render(<App/>);

    expect(screen.getByTestId('counter-result')).toHaveTextContent('Counter: 0')

    screen.getByTestId('plus').click()

    expect(screen.getByTestId('counter-result')).toHaveTextContent('Counter: 1')
  })

  it('decreases the counter', () => {})

  it('displays the emoji when counter < 0', () => {
    render(<App/>);

    expect(screen.getByTestId('counter-result')).toHaveTextContent('Counter: 0')

    screen.getByTestId('minus').click()

    const actual = screen.getByTestId('counter-result');
    const expected = ':-)';

    expect(actual).toHaveTextContent(expected)
  })

})