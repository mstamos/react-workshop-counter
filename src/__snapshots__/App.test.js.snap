// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Buttons renders 1`] = `
<div>
  <div>
    <button
      data-testid="plus"
    >
      +
    </button>
    <button
      data-testid="minus"
    >
      -
    </button>
  </div>
</div>
`;
